### DISCUSSION TITLE
Repository (Management) Discussions

### DISCUSSION THREAD

**Discussion by nobkd** [nuejs/nue#211] [General]
I'd collect general things about the repository and repository management here, that (I feel) are worth a discussion.

I think it would be okay to use each "main answer" under this discussion as a new starting point for topics, so the talk doesn't spread out over different issues, PRs, other discussions.
(This can also be a negative point, if there are too many topics, but I think it will be okay for now)

Feel free to open your own related topic here

**Comment by nobkd** [comment]
# More labels for issues / discussions / PRs

## package

I would like labels for each package of Nue, so it's clearly visible which e.g. feature request applies to what package(s).

So `glow`, `core`, `mark`, `kit` or similar.

## state

Labels that show the state of an issue / discussion / PR.
E.g. `needs discussion`, `needs decision`, `needs docs`, `needs tests`, etc.

cc @goblinfactory with https://github.com/nuejs/nue/issues/198#issuecomment-1943631199

(Maybe those also without `needs` / `requires`, like `docs` / `tests`, to show what an issue is about without showing the need to change something.)

## other

`needs help` / `help needed` / `looking for help` / `help`, ...

---

**Edit**: I added some labels:
- `needs discussion`: Needs a discussion or decision
- `tests`
- `nuemark`: Related to[ nuemark ](https://github.com/nuejs/nue/labels/nuemark)package
- `nuekit`: Related to nuekit package
- `nuejs-core`: Related to[ nuejs-core ](https://github.com/nuejs/nue/labels/nuejs-core)package
- `nue-glow`: Related to nue-glow package
- `docs` (Because docs seems to be included in packages in the future? and also to refer to docs when needed)

**Comment by nobkd** [comment]
# Merging PRs and Access Management

I think @tipiirai didn't get the notification for this: https://github.com/nuejs/nue/pull/155#issuecomment-1935224380

We currently can not merge PRs as `triage` permission does not allow that.
See: https://docs.github.com/en/organizations/managing-user-access-to-your-organizations-repositories/managing-repository-roles/repository-roles-for-an-organization

So if you asked yourself why we (@fritx, I) didn't merge anything until now, now you know why 😉

---

**Edit**: at least I (not sure about `@fritx`) have maintainer role now. Ref: https://github.com/nuejs/nue/pull/276#issuecomment-2134346898 (hiding this as resolved for now)

**Comment by nobkd** [comment]
# JS Docstrings

See #201

(Could be relevant for the future development.)

Yes / No / Maybe in the future, but not now

**Reply by tipiirai** [reply]
I vote no. I suggest we put this effort to improving test suites. Time better spent I think.

**Reply by tipiirai** [reply]
On the next update I'm adding clarity to how types should be dealt in web application development and how TypeScript is placed in the ecosystem.

**Comment by nobkd** [comment]
# Lock-file handling

## Lock files

### Pro

- Reproducible tests (if used with frozen-lock file on installations)
  https://github.com/nuejs/nue/pull/150#issuecomment-1875224333
   https://github.com/nuejs/nue/pull/154#discussion_r1446710714
- Faster installs (at least for bun)
  https://github.com/nuejs/nue/pull/150#issuecomment-1919982493
  - does not have to rebuild the dependency tree all the time

### Contra / Needs decision

- binary lock file is unreadable unless using an extension / updating the git config
  https://github.com/nuejs/nue/pull/149#issuecomment-1921662804
- problem of "least surprise" (scary changed binary lock file / git conflicts)
  https://github.com/nuejs/nue/pull/149#issuecomment-1922086072
  https://github.com/nuejs/nue/pull/150#issuecomment-1875262328
- no / one / multiple lock files
  https://github.com/nuejs/nue/pull/154#discussion_r1446768158
- might need frequent updates
  https://github.com/nuejs/nue/pull/149#issuecomment-1921907957

**Reply by nobkd** [reply]
Update: bun now has a text based lock file as default. We could migrate to it.

**Reply by tipiirai** [reply]
I'm a rookie when it comes to lock files, so not much opinions here. But I definitely favor Bun first approach. With CSS support, I might even vote for dropping Node support completely.

**Comment by tipiirai** [comment]
@nobkd you are clearly a key contributor to Nue. I'd like to talk about things. What is your timetable next week? Can you send me a calendar invite anywhere between 8-20 EET time?
